SudokuFuncs.py

```diff
@@ -188,63 +186,26 @@ def list_difference(li1, li2):
 
     returns: resulting list after subtraction
     """
-    li_dif = [i for i in li1 + li2 if i not in li1 or i not in li2]
+    li_dif = [i for i in li1 + li2 if i in li1 and i not in li2]
     return li_dif
```

Comment:
```suggestion
    return [i for i in li1 + li2 if i in li1 and i not in li2]
```
Let's not waste memory by specifically allocating the list to a variable and then returning it ;)
Make changes for any other similar snippets in the program, if found.